### PR TITLE
fix: #2272, #2273 - create correct policies when IAM is the default auth

### LINKED ISSUE #2272: Policy statement must contain resources after upgrade to 3.2 and add @auth
1. Upgraded from 3.0 to 3.2
2. Push failed (defaultAuthorization not set)
3. Ran update api
4. Push succeeds
5. Added @auth(rules:[{allow:public, provider:iam, operations:[read]}])
6. Push fails with:

Resource Name: theme-Unau-ES6W4PH9WDDZ (AWS::IAM::Policy)
Event Type: create
Reason: Policy statement must contain resources. (Service: AmazonIdentityManagement; Status Code: 400; Error Code: MalformedPolicyDocument; Request ID: 0dbcd5ff-d5ca-11e9-93c6-3da30b79754d)

7. Removed aim from above to @auth(rules:[{allow:public, operations:[read]}])
8. Push succeeds...but is not what I need...

### PR DESCRIPTION
**Pull request by attilah**
*Issue #, if available:*

Fixes #2272, #2273

*Description of changes:*

- When IAM is the default authentication mode, an empty policy was generated, because @aws_iam directives were not added, so it did not add the policy resource.
- @key generated queries were not properly protected, they missed the directives and the IAM policy resource entries.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/graphql-auth-transformer/src/ModelAuthTransformer.ts [hunk 1]
```diff
@@ -431,28 +442,33 @@ Static group authorization should perform as expected.`
...
                     }
                 }
 
                 addDirectivesForOperation('get');
                 addDirectivesForOperation('list');
             }
 
+            if (operationName) {
+                this.addFieldToResourceReferences(ctx.getQueryTypeName(), operationName, rules);
+                this.addFieldToResourceReferences(ctx.getQueryTypeName(), operationName, rules);
```
Comment (kaustavghosh06) [inline packages/graphql-auth-transformer/src/ModelAuthTransformer.ts]: Why do we have the same statement twice?

**Comment by kaustavghosh06** [comment]
Could you also please add an e2e test with @key.

File: packages/graphql-auth-transformer/src/ModelAuthTransformer.ts (hunk 1 above)
Comment (attilah) [reply packages/graphql-auth-transformer/src/ModelAuthTransformer.ts]: Already refactored, that piece was not correct there, victim of copy and paste.

**Comment by attilah** [comment]
> Could you also please add an e2e test with @key.

I plan to add tests as well, to make sure we're covering the grounds.

**Pull request merged**